transaction/gas/node_station.go

```diff
@@ -0,0 +1,49 @@
+package gas
+
+import (
+	"github.com/ethereum/go-ethereum/consensus/misc"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/params"
+	"math/big"
+)
+
+type NodeStation struct {
+	bc      BCClient
+	chainID int64
+}
+
+func NewNodeStation(bc BCClient, chainID int64) *NodeStation {
+	return &NodeStation{bc: bc, chainID: chainID}
+}
+
+type BCClient interface {
+	SuggestGasPrice(chainID int64) (*big.Int, error)
+	HeaderByNumber(chainID int64, number *big.Int) (*types.Header, error)
+	BlockNumber(chainID int64) (uint64, error)
+}
+
+func (n *NodeStation) GetGasPrices() (*GasPrices, error) {
+	suggestGasPrice, err := n.bc.SuggestGasPrice(n.chainID)
+	if err != nil {
+		return nil, err
+	}
+
+	blockNumber, err := n.bc.BlockNumber(n.chainID)
```

Comment:
no need for this, you can pass nil https://pkg.go.dev/github.com/ethereum/go-ethereum/ethclient#Client.HeaderByNumber